libs/zephyr-agent/src/lib/build-context/ze-styled-prompts.ts

```diff
@@ -0,0 +1,81 @@
+import * as readline from 'node:readline';
+import { logFn } from '../logging/ze-log-event';
+import { bold, bgGreenBright, black } from '../logging/picocolor';
+
+interface PromptOptions {
+  message: string;
+  placeholder?: string;
+  defaultValue?: string;
+  validate?: (value: string) => string | undefined;
+}
+
+/** Custom prompt implementation that matches Zephyr's visual style */
+export async function zephyrPrompt(options: PromptOptions): Promise<string> {
+  const { message, placeholder, defaultValue, validate } = options;
+
+  // Display the prompt message with Zephyr styling
+  process.stdout.write('\n');
+  logFn('info', message);
+
+  const displayPlaceholder = placeholder || defaultValue || '';
+  const promptPrefix = bold(bgGreenBright(black(' PROMPT ')));
+
+  const rl = readline.createInterface({
+    input: process.stdin,
+    output: process.stdout,
+  });
+
+  return new Promise((resolve, reject) => {
+    // Create the prompt with default value inline
+    const promptText = displayPlaceholder
+      ? `${promptPrefix}  (${displayPlaceholder}) > `
+      : `${promptPrefix}  > `;
+
+    rl.question(promptText, (input) => {
+      const value = input.trim() || defaultValue || '';
+
+      if (validate) {
+        const error = validate(value);
+        if (error) {
+          logFn('error', error);
+          rl.close();
+          // Recursively call for retry
+          zephyrPrompt(options).then(resolve).catch(reject);
+          return;
+        }
+      }
+
+      rl.close();
+      resolve(value);
+    });
+
+    rl.on('SIGINT', () => {
+      console.log(); // New line before the warning
+      logFn('warn', 'Operation cancelled by user');
+      rl.close();
+      process.exit(1);
```

Comment:
```suggestion
```

I don't agree with this exit here. It should not terminate the user's build but instead just fail to upload it to zephyr or try something else.